ironfish/src/utils/array.test.ts

```diff
@@ -0,0 +1,43 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { ArrayUtils } from './array'
+
+describe('ArrayUtils', () => {
+  it('shuffles array', () => {
+    let items: number[] = []
+
+    for (let i = 0; i <= 10000; ++i) {
+      items.push(i)
+    }
+
+    items = ArrayUtils.shuffle(items)
+
+    const shuffled = items[0] !== 0 && items[500] !== 500 && items[1000] !== 1000
```

Comment:
Nit: this test would pass if all elements are set to 99. In that case, the array wouldn't be shuffled, but filled with copies of a single element.

Probably a better test would be to check that it contains all the original items, but in a different order, like this (untested):

```ts
const shuffled = ArrayUtils.shuffle(items)
expect(shuffled).not.toEqual(items)
expect(shuffled.sort()).toEqual(items)
```

Still not perfect, but a bit more robust.

Reply:
The test doesn't set all elements to 99, the test relies on every element being different.

However I still think your test is better so I'm gonna use that.

Reply:
Fixed